src/test/java/net/revelc/code/formatter/AbstractFormatterTest.java

```diff
@@ -73,16 +73,19 @@ public String getCompilerCodegenTargetPlatform() {
         }
     }
 
-    protected void doTestFormat(Formatter formatter, String fileUnderTest, String expectedSha512) throws IOException {
+    protected void doTestFormat(Map<String, String> options, Formatter formatter, String fileUnderTest,
+            String expectedSha512) throws IOException {
+
         File originalSourceFile = new File("src/test/resources/", fileUnderTest);
         File sourceFile = new File("target/testoutput/", fileUnderTest);
+        Files.copy(originalSourceFile, sourceFile);
 
-        Map<String, String> options = new HashMap<>();
+        if (null == options) {
```

Comment:
This isn't necessary if what is passed in is already an empty set instead of null. Since this is test code, we already control that... so no need to write protectively like this. Better to be consistent in the test code, and pass in the options in the expected format to the test method. If you want to make sure we didn't write bad test code, could use JUnit's assert: `Assert.assertNotNull(options);`

Reply:
This check was necessary to ensure options were not getting overwritten to a blank set. See above comment - https://github.com/revelc/formatter-maven-plugin/pull/280/files/d622427c339ba6d4f17748ce89d89ca5f27d33b4#r191646599

Reply:
My point was that you can make that a prerequisite which is asserted to be true instead of a conditional. Then, make sure any caller provides their own non-null set so they don't fail the assertion.